pelican_bib/bib.py

```diff
@@ -160,6 +201,159 @@ def add_publications(generator):
     generator.context['publications_lists'] = publications_lists
 
 
+class Bibliography(Directive):
+    """
+    Directive to embed bibliographies into articles/posts.
+
+    Usage::
+
+        .. bibliography:: PUBLICATIONS_SRC
+          :template: TEMPLATE_NAME (optional)
+          :options: TEMPLATE_OPTIONS_DICT (optional)
+          :class: CLASS_NAME (optional)
+          :filter_tag: FILTER_TAG (optional)
+          :pybtex_style_args: PYBTEX_STYLE_ARGS (optional)
+          :sorting_style: PYBTEX_SORTING_STYLE (optional)
+          :abbreviate_names: PYBTEX_ABBREVIATE_NAMES (optional)
+          :name_style: PYBTEX_NAME_STYLE (optional)
+
+    e.g.::
+
+        .. bibliography:: osm.bib
+          :template: publications_unsrt
+          :options: {'groupby_value': year}
+          :class: my-bib
+          :pybtex_style_args: {'sorting_style': 'author_year_title',
+                               'abbreviate_names': False,
+                               'name_style': 'lastfirst' }
+          :sorting_style: author_year_title
+          :abbreviate_names: False
+          :name_style: lastfirst
+    """
+    required_arguments = 0
+    optional_arguments = 100
+    final_argument_whitespace = False
+    has_content = True
+
+    generator = None
+
+    def set_generator(generator):
+        Bibliography.generator = generator
+
+    def boolean(argument):
+        if argument == 'True': return True
+        if argument == 'False': return False
+        raise ValueError('must be boolean')
```

Comment:
Does it suffice to use the built-in function `isinstance` maybe?

```suggestion
    def boolean(argument):
        if not isinstance(argument, (bool)):
            raise TypeError()
```

Reply:
`argument` is always `<class 'str'>` so your solution wouldn't work.
 
But I replaced the custom `boolean` spec with `distutils.util.strtobool`